grype/db/internal/gormadapter/open.go

```diff
@@ -4,37 +4,81 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/glebarez/sqlite"
 	"gorm.io/gorm"
+
+	"github.com/anchore/grype/internal/log"
 )
 
+var commonStatements = []string{
+	`PRAGMA foreign_keys = ON`, // needed for v6+
+}
+
 var writerStatements = []string{
-	// performance improvements (note: will result in lost data on write interruptions).
-	// on my box it reduces the time to write from 10 minutes to 10 seconds (with ~1GB memory utilization spikes)
-	`PRAGMA synchronous = OFF`,
-	`PRAGMA journal_mode = MEMORY`,
-	`PRAGMA cache_size = 100000`,
-	`PRAGMA mmap_size = 268435456`, // 256 MB
+	// performance improvements (note: will result in lost data on write interruptions)
+	`PRAGMA synchronous = OFF`,     // minimize the amount of syncing to disk, prioritizing write performance over durability
+	`PRAGMA journal_mode = MEMORY`, // do not write the journal to disk (maximizing write performance); OFF is faster but less safe in terms of DB consistency
 }
 
-var readOptions = []string{
-	"immutable=1",
-	"cache=shared",
-	"mode=ro",
+var heavyWriteStatements = []string{
```

Comment:
nit: this doesn't seem to have options directly related to write performance but rather memory, with an option named `allowLargeMemoryFootprint` should this be `largeMemoryFootprintStatements`?

Reply:
https://github.com/anchore/grype/pull/2335#discussion_r1887662856 so when there are heavy writes expected we want to allow for a larger memory footprint (to improve write performance).